server/middlewares/article.middlewares.js

```diff
@@ -97,11 +97,99 @@ const validateHighlight = (req, res, next) => {
   return next();
 };
 
+const reviewArticleValidator = (req, res, next) => {
+  const { articleId } = req.params;
+  if (!Object.keys(req.body).length) {
+    return res.status(400).json({
+      errors: {
+        body: ['No input provided'],
+      },
+    });
+  }
+
+  if (!validations.verifyUUID(articleId)) {
+    return res.status(400).json({
+      errors: {
+        body: ['id not valid'],
+      },
+    });
+  }
+
+  if (!req.user.userObj.isReviewer) {
+    return res.status(403).json({
+      errors: {
+        body: ['User is not a reviewer'],
+      },
+    });
+  }
+
+  if (!req.body.reviewer_comment) {
+    return res.status(400).json({
+      errors: {
+        body: ['Reviewer comment is required'],
+      },
+    });
+  }
+
+  return next();
+};
+
+const articleStatusValidator = (req, res, next) => {
+  const { articleId } = req.params;
+  if (!Object.keys(req.body).length) {
+    return res.status(400).json({
+      errors: {
+        body: ['No input provided'],
+      },
+    });
+  }
+
+  if (!validations.verifyUUID(articleId)) {
+    return res.status(400).json({
+      errors: {
+        body: ['id not valid'],
+      },
+    });
+  }
+
+  if (!req.user.userObj.isAdmin) {
+    return res.status(401).json({
+      errors: {
+        body: ['User is not an admin'],
```

Comment:
This should return a 403 error code @sojida

Reply:
Although it depends on the action - if an authenticated user is trying to do something that only admins should do then it should definitely be a 403

Reply:
It is definitely a 403. I'll update this ASAP.